CSVSource/README.md

```diff
@@ -0,0 +1,157 @@
+6.  Run `<install location>/CSVSource/bin/CSVSource` with a local server.config file or specifying the [server config file](#serverConfig) as the first argument.
+
+## Logging
+To change the logging settings, edit the logging config file `<install location>/CSVSource/src/main/resources/log4j2.xml`,
+which is an [Apache Log4j configuration file.](https://logging.apache.org/log4j/2.x/manual/configuration.html). The logger 
+name for each class is the class's fully qualified class name, *e.g.* "io.vantiq.extjsdk.ExtensionWebSocketClient".  
+
+## Server Config File
+
+The server config file is written as `property=value`, with each property on its
+own line. The following is an example of a valid server.config file:
+```
+authToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+sources=CSV1
+targetServer=https://dev.vantiq.com/
+```
+
+### Vantiq Options
+*   **authToken**: Required. The authentication token to connect with. These can be obtained from the namespace admin.
+*   **sources**: Required. A comma separated list of the sources to which you wish to connect. Any whitespace will be
+    removed when read.
+*   **targetServer**: Required. The Vantiq server hosting the sources.
+
+# Setting Up VANTIQ <a name="vantiq" id="vantiq"></a>
+
+## Source Configuration
+
+To set up the Source in the VANTIQ Modelo IDE, you will need to add a Source to your project. Make sure you have properly added a Source Definition to VANTIQ. Once this is complete, you can select CSV (or whatever you named your Source Definition) as the Source Type. You will then need to fill out the Source Configuration 
+Document.
+
+The Configuration document may look similar to the following example:
+
+    {
+        "csvConfig": {
+        "fileFolderPath": "d:/tmp/csv",
+        "filePrefix": "eje",
+        "fileExtension": "csv",
+        "maxLinesInEvent": 200,
+        "delimiter":",",
+            "schema": {
+                "field0": "value",
+                "field2": "flag",
+                "field1": "YScale"
+            }
+        },
+        "options": {
+            "maxActiveTasks": 2,
+            "maxQueuedTasks": 4,
+            "processExistingFiles": true,
+            "extensionAfterProcessing": "csv.done",
+            "deleteAfterProcessing": false
+        }
+    }
+
+### Configuration
+*   **fileFolderPath**: Required. The folder in which this source will look for CSV files. 
+*   **filePrefix**: Optional . The prefix of the file pattern to look for, if not set any file name will be accepted. 
+*   **fileExtension**: Required. The file extension of the files to be processed 
+*   **maxLinesInEvent**: Required. Determine how many lines from the CSV file will be sent in a single message to the server. Depending on the number of the lines of the CSV file, a high value might result in messages too large to process efficiently or a memory exception. 
+*   **delimiter**: the delimiter to be used when parse the CSV file
```

Comment:
Need to describe what this is.  It's a regular expression that separates entries in the files being processed.  The default value is "," or whatever (should be ",").

Reply:
done